airbyte-workers/src/main/java/io/airbyte/workers/internal/EmptyAirbyteSource.java

```diff
@@ -43,11 +103,44 @@ public int getExitValue() {
 
   @Override
   public Optional<AirbyteMessage> attemptRead() {
-    if (!hasEmittedState.get()) {
-      hasEmittedState.compareAndSet(false, true);
-      return Optional.of(new AirbyteMessage().withType(Type.STATE).withState(new AirbyteStateMessage().withData(Jsons.emptyObject())));
+    if (!isStarted) {
+      throw new IllegalStateException("The empty source has not been started.");
+    }
+
+    if (isPartialReset) {
```

Comment:
these conditionals are hard to follow. can you break them into smaller methods please?

Reply:
Done